### PR TITLE
Add AWS delete protection

### PR DESCRIPTION
**Pull request by blacksails**
This adds the AWS delete protection which ensures that the database is not accidentally deleted within AWS.

### CODE REVIEW

File: rds/rds_provider.go [hunk 1]
```diff
@@ -251,6 +251,7 @@ func convertSpecToInput(v *crd.Database, subnetName string, securityGroups []str
 		MultiAZ:               aws.Bool(v.Spec.MultiAZ),
 		StorageEncrypted:      aws.Bool(v.Spec.StorageEncrypted),
 		BackupRetentionPeriod: aws.Int64(v.Spec.BackupRetentionPeriod),
+		DeletionProtection:    aws.Bool(v.Spec.DeleteProtection),
```
Comment (domask) [inline rds/rds_provider.go]: will it apply for the existing RDS instances?

Comment (blacksails) [reply rds/rds_provider.go]: no the rds-operator currently does not support changes of the crd.

Comment (sorenmat) [reply rds/rds_provider.go]: I guess we could add it??

Comment (sorenmat) [reply rds/rds_provider.go]: Let's do it in another PR though :)

Comment (blacksails) [reply rds/rds_provider.go]: IMO creation and updates should run the same code, checking status of resources and reconciling. Like kubebuilder <3

**Pull request merged**